### PR TITLE
use new evaluate resource endpoint

### LINKED ISSUE #13: update enforcer to use new resource evaluation RPC

### PR DESCRIPTION
**Pull request by mrparkers**
Closes #13

### CODE REVIEW

File: enforcer/enforcer.go [hunk 1]
```diff
@@ -198,41 +199,66 @@ func (e *Enforcer) evaluatePolicy(log *zap.Logger, response *v1.AdmissionRespons
...
+func (e *Enforcer) getEvaluationSummary(ctx context.Context, imageName string, resourceEvaluationResult *rode.ResourceEvaluationResult) (string, error) {
+	var sb strings.Builder
 
-	if err != nil {
-		log.Error("failed to retrieve policy", zap.Error(err))
-		return ""
+	sb.WriteString(fmt.Sprintf(`container image "%s" %s evaluation against the rode policy group "%s": `, imageName, policyResult(resourceEvaluationResult.ResourceEvaluation.Pass), resourceEvaluationResult.ResourceEvaluation.PolicyGroup))
+
+	for i, policyEvaluation := range resourceEvaluationResult.PolicyEvaluations {
+		policy, err := e.rode.GetPolicy(ctx, &rode.GetPolicyRequest{
```
Comment (mrparkers) [inline enforcer/enforcer.go:237]: we're probably going to include the policy name within these policy evaluation details, so this shouldn't be necessary in the future

File: enforcer/enforcer.go [hunk 2]
```diff
@@ -198,41 +199,66 @@ func (e *Enforcer) evaluatePolicy(log *zap.Logger, response *v1.AdmissionRespons
...
 	}
 
 	return false
 }
 
-func (e *Enforcer) getPolicyName(log *zap.Logger) string {
-	policy, err := e.rode.GetPolicy(context.Background(), &rode.GetPolicyRequest{
-		Id: e.config.PolicyId,
-	})
+func (e *Enforcer) getEvaluationSummary(ctx context.Context, imageName string, resourceEvaluationResult *rode.ResourceEvaluationResult) (string, error) {
```
Comment (mrparkers) [inline enforcer/enforcer.go:231]: here's what the message looks like:

```
container image "harbor.parker.gg/rode-demo/rode-demo-node-app:fail-sonar-policy-0ed7b90" failed evaluation against the rode policy group "test": policy "Sample Harbor Policy" PASSED | policy "Sample SonarQube Policy" FAILED
```

File: enforcer/enforcer.go [hunk 3]
```diff
@@ -160,7 +160,8 @@ func (e *Enforcer) fetchImagePullSecrets(pod *corev1.Pod) ([]*imagePullSecret, e
 }
 
 func (e *Enforcer) evaluatePolicy(log *zap.Logger, response *v1.AdmissionResponse, imageName string, pullSecrets []*imagePullSecret) bool {
-	log = log.With(zap.String("image", imageName))
+	ctx := context.Background()
+	log = log.With(zap.String("image", imageName)).With(zap.String("policy group", e.config.PolicyGroup))
```
Comment (alexashley) [inline enforcer/enforcer.go]:
```suggestion
	log = log.With(zap.String("image", imageName)).With(zap.String("policyGroup", e.config.PolicyGroup))
```

Nitpick, but with the production config the log format will be json and it's a little awkward to have the space there, for instance when writing log queries.

File: enforcer/enforcer.go [hunk 4]
```diff
@@ -198,41 +199,66 @@ func (e *Enforcer) evaluatePolicy(log *zap.Logger, response *v1.AdmissionRespons
...
-	policy, err := e.rode.GetPolicy(context.Background(), &rode.GetPolicyRequest{
-		Id: e.config.PolicyId,
-	})
+func (e *Enforcer) getEvaluationSummary(ctx context.Context, imageName string, resourceEvaluationResult *rode.ResourceEvaluationResult) (string, error) {
+	var sb strings.Builder
 
-	if err != nil {
-		log.Error("failed to retrieve policy", zap.Error(err))
-		return ""
+	sb.WriteString(fmt.Sprintf(`container image "%s" %s evaluation against the rode policy group "%s": `, imageName, policyResult(resourceEvaluationResult.ResourceEvaluation.Pass), resourceEvaluationResult.ResourceEvaluation.PolicyGroup))
```
Comment (alexashley) [inline enforcer/enforcer.go]:
```suggestion
	sb.WriteString(fmt.Sprintf(`container image "%s" %s evaluation against the Rode policy group "%s": `, imageName, policyResult(resourceEvaluationResult.ResourceEvaluation.Pass), resourceEvaluationResult.ResourceEvaluation.PolicyGroup))
```

I know we usually stylize the name as lowercase, but I think in this context it's clearer if it's uppercase.

**Review by alexashley — APPROVED** [review]
LGTM, tested it out locally. Just some nits that you can address or ignore.

**Pull request merged**